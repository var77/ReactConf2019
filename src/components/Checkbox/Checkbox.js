import React from 'react';

const Checkbox = ({ checked, onCheck }) => {
    return <input type="checkbox" checked={checked} onChange={onCheck} />;
};

export default Checkbox;