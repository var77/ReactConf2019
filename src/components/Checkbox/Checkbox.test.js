import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from './Checkbox';

describe('Checkbox Component', () => {
    it ('should render without crashing', () => {
        const el = shallow(<Checkbox />);
        expect(el).toMatchSnapshot();
    });

    it ('should check the checkbox', () => {
        let checked = false, el = null;
        const onCheck = jest.fn(() => el.setProps({ checked: !el.prop('checked') }));
        el = shallow(<Checkbox onCheck={onCheck} checked={checked} />);
        expect(el.prop('checked')).toBe(false);
        el.simulate('change');
        expect(onCheck).toHaveBeenCalledTimes(1);
        expect(el.props().checked).toBe(true);
    });
});