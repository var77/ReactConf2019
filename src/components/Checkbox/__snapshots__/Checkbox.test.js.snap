// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should render without crashing 1`] = `
<input
  type="checkbox"
/>
`;
